packages/@sanity/components/src/edititem/PopOver.js

```diff
@@ -175,259 +108,137 @@ export default class EditItemPopOver extends React.Component {
-      modalTranslateY = scrollTop - newScrollTop + currentModalTranslateY
-      this._portalModalElement.style.transform = `translateY(${modalTranslateY}px)`
-
-      // Do the scroll
-      scroll.top(scrollContainer, newScrollTop, scrollOptions, this.addMovingListeners)
-
-    } else {  // Model content fits within the current screen
-      this._contentElement.className = styles.content
-      // Set the new translate
-      modalTranslateY = scrollTop - this._initialScrollTop + currentModalTranslateY
-      this._portalModalElement.style.transform = `translateY(${modalTranslateY}px)`
-      // Do the scroll
-      scroll.top(scrollContainer, this._initialScrollTop, scrollOptions, this.addMovingListeners)
+    if (availableWidth < (rootLeft + (width / 2))) {
+      popoverLeft = availableWidth - width - PADDING
     }
 
-    // Move other modals accordingly (on open)
-    if (shouldMoveOtherModals) {
-      popOverStack.slice(0, -1).forEach(popOver => {
-        const translateY = getComputedTranslateY(popOver._portalModalElement)
-        popOver._translateYHistory.push(translateY)
-        const newTranslateY = translateY + modalTranslateY
-        popOver._portalModalElement.style.transform = `translateY(${newTranslateY}px)`
-      })
-    }
-  }
-
-  handlePortalOpened = element => {
-    this.moveIntoPosition(true)
-  }
-
-  setArrowElement = element => {
-    this._arrowElement = element
-  }
-
-  setPortalModalElement = element => {
-    this._portalModalElement = element
-  }
-
-  setContentElement = element => {
-    this._contentElement = element
-  }
-
-  setRootElement = element => {
-    this._rootElement = element
-    const {isOpen} = this.props
-    this.setState({portalIsOpen: isOpen})
+    this.setState({
+      popoverLeft: popoverLeft,
+      availableHeight: availableHeight,
+      arrowLeft: rootLeft,
+      isResizing: isScrolling
+    })
   }
 
-  handleContentScroll = event => {
-    this._contentScrollTop = event.target.scrollTop
-  }
+  render() {
+    const {
+      title,
+      children,
+      actions,
+      isOpen,
+    } = this.props
+
+
+    const {
+      popoverLeft,
+      arrowLeft,
+      availableHeight,
+      scrollContainer,
+      isResizing
+    } = this.state
 
-  renderPortal = () => {
-    const {title, children, className, isCreatingNewItem, actions, fullWidth} = this.props
-    const modalContainerClassName = `${fullWidth ? styles.fullWidth : styles.autoWidth} ${className || ''}`
     return (
-      <Portal isOpened={this.state.portalIsOpen} closeOnEsc={false} onOpen={this.handlePortalOpened}>
-        <div className={modalContainerClassName}>
-          {
-            this.state.isFocused && (
-              <div className={styles.overlay} onClick={this.handleBackdropClick} />
-            )
-          }
+      <div style={{display: 'span'}} ref={this.setRootElement}>
+        <StickyPortal
+          isOpen={isOpen}
+          scrollContainer={scrollContainer}
```

Comment:
Why do we need to pass this down? If StickyPortal needs some values from the scrollContainer, its a lot better to pass down these values instead.

Reply:
It needs to add an eventListener and get the positions. I think that an ref to an element is a good way to go. 

I dont like the "tryFindScrollContainer" way of doing this. In the future it would be nice to control this by passing down props from where the scrollContainer is (desk-tool in this case), or have an global system to control modals/overlay. This is also good for determine what should be closed on pressing escape etc.

Reply:
Yeah, lets try to fix this properly later.